editor/src/components/canvas/canvas-strategies/absolute-move-strategy.ts

```diff
@@ -38,27 +45,58 @@ export const absoluteMoveStrategy: CanvasStrategy = {
       interactionState.interactionData.drag != null
     ) {
       const drag = interactionState.interactionData.drag
-      return canvasState.selectedElements.flatMap((selectedElement) => [
-        adjustNumberProperty(
-          'permanent',
-          selectedElement,
-          stylePropPathMappingFn('left', ['style']),
-          drag.x,
-          true,
-        ),
-        adjustNumberProperty(
-          'permanent',
-          selectedElement,
-          stylePropPathMappingFn('top', ['style']),
-          drag.y,
-          true,
-        ),
+      return [
+        ...canvasState.selectedElements.flatMap((selectedElement) => {
+          const element: JSXElement | null = withUnderlyingTarget(
+            selectedElement,
+            canvasState.projectContents,
+            {},
+            canvasState.openFile,
+            null,
+            (_, e) => e,
+          )
+          const elementParentBounds =
+            MetadataUtils.findElementByElementPath(
+              sessionState.startingMetadata, // TODO should this be using the current metadata?
+              selectedElement,
+            )?.specialSizeMeasurements.immediateParentBounds ?? null
+
+          if (element == null) {
+            return []
+          }
+
+          return mapDropNulls(
+            (pin) => {
+              const horizontal = isHorizontalPoint(
+                // TODO avoid using the loaded FramePoint enum
+                framePointForPinnedProp(pin),
+              )
+              const negative = pin === 'right' || pin === 'bottom'
+              const value = getLayoutProperty(pin, right(element.props), ['style'])
+              if (isRight(value) && value.value != null) {
+                // TODO what to do about missing properties?
```

Comment:
In theory we should insert them based on the computed values in the metadata, but I question whether that's something that should happen in this strategy, or in a separate strategy that exists purely for the purpose of inserting a missing layout prop (I'm thinking along the lines of the "convert to absolute" strategy we have spoken about)

Reply:
I should clarify that this is a Design TODO

Reply:
added todo to doc